packages/flutter_tools/test/general.shard/generate_localizations_test.dart

```diff
@@ -1598,30 +1598,23 @@ import 'output-localization-file_en.dart' deferred as output-localization-file_e
           ..createSync(recursive: true);
         l10nDirectory.childFile(defaultTemplateArbFileName)
           .writeAsStringSync(selectMessageWithoutPlaceholdersAttribute);
-
-        expect(
-          () {
-            LocalizationsGenerator(
-              fileSystem: fs,
-              inputPathString: defaultL10nPathString,
-              outputPathString: defaultL10nPathString,
-              templateArbFileName: defaultTemplateArbFileName,
-              outputFileString: defaultOutputFileString,
-              classNameString: defaultClassNameString,
-              logger: logger,
-            )
-              ..loadResources()
-              ..writeOutputFiles();
-          },
-          throwsA(isA<L10nException>().having(
-            (L10nException e) => e.message,
-            'message',
-            contains('''
-Make sure that the specified placeholder is defined in your arb file.
-[app_en.arb:helloWorld] Hello {name}
-                               ^'''),
-          )),
-        );
+        LocalizationsGenerator(
+          fileSystem: fs,
+          inputPathString: defaultL10nPathString,
+          outputPathString: defaultL10nPathString,
+          templateArbFileName: defaultTemplateArbFileName,
+          outputFileString: defaultOutputFileString,
+          classNameString: defaultClassNameString,
+          logger: logger,
+        )
+          ..loadResources()
+          ..writeOutputFiles();
+        expect(logger.hadWarningOutput, isTrue);
+        expect(logger.warningText, contains('''
+[app_en.arb:helloWorld] ICU Syntax Warning: Missing placeholder declaration. Defaulting to using the message as is without parsing.
```

Comment:
Can you check the full error messages in these tests? The issue mentions parts of the error message missing. You could probably define a template for the message and re-use it to simplify the test code here.

Reply:
From the issue

> The tests that do exist, don't seem to be testing that the second two lines of the error message print something useful.
The second part of the message is coming through empty for me, and the third part, which is supposed to have the file and line number are empty, so I can't track down the issue with flutter/gallery.

Reply:
I wasn't able to reproduce that behavior, but the test is now checking for the full error message. I also ran `flutter run` in `flutter/gallery` and was able to see the warnings.